py4j-python/src/py4j/java_gateway.py

```diff
@@ -273,6 +273,7 @@ def launch_gateway(port=0, jarpath="", classpath="", javaopts=[],
         directory of the Java process.
     :param return_proc: If True, returns the Popen object returned when the JVM
         process was created.
+    :param use_shell – If True, Popen will be start the java process with shell=True
```

Comment:
this parameter should be documented and added to the launch_gateway classmethod in the JavaGateway class

Reply:
I have updated code to use 
    popen_kwargs["shell"] = use_shell 
instead as you wanted (the reason I did it like it did before was that i did not want to cause any strange side-effects when using the default value)

I have also changed the class method.

Reply:
Thank you. I also hesitated, but shell=False is supposed to be the default value on all OSes so hopefully all python interpreters follow the API documentation :crossed_fingers: